### PR TITLE
New Calendar component

### PR DESCRIPTION
**Pull request by cerinman**
This PR extracts the calendar logic from the `DatePicker` component to it's own `Calendar` component.  I've done this because we now have duplicate code for the calendar in 5 locations between the open source and internally.  I don't want to maintain all that so here we go.

This PR does the following:

- Extracts the calendar code to it's own component as already noted above.
- Refactors the `DatePicker` component to use the new `Calendar` component.
- Sets up the Demo app to use the new `Calendar` component.

I'll update the `DatePickerFullScreen` and new `DateRangePicker` components in separate PRs after this one gets merged.  I didn't do that here because the `DatePickerFullScreen` component is way behind the curve for being up to date and the new `DateRangePicker` needs to be able to select ranges so I'll have to read through it's code a bit more before I implement the `Calendar` component there.

#### Screen

screen shot 2016-06-13 at 5 10 35 pm
screen shot 2016-06-13 at 5 11 04 pm

### CODE REVIEW

**Comment by jmophoto** [comment]
🍪  for this. Will use it in the date time picker! :shipit:

File: src/components/Calendar.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  },
+
+  getInitialState () {
+    return {
+      currentDate: this.props.selectedDate || moment().unix()
+    };
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.selectedDate && newProps.selectedDate !== this.props.selectedDate) {
```
Comment (redperk) [inline src/components/Calendar.js]: I'm confused on why the same thing is written twice. Is this correct or is it supposed to be `if (newProps && newProps.selectedDate) .....`?

**Comment by iheartrachie** [comment]
I'm all for DRYing up code as much as possible. This is already merged and this should work without issue for `DatePickerFullScreen` but I'm not sure I see how it can be cohesive with `DateRangePicker`. It has slightly different props but vastly different CSS and logic in the component.

File: src/components/Calendar.js (hunk 1 above)
Comment (redperk) [inline src/components/Calendar.js]: or wait, is it checking first to make sure it's not undefined and if it's not then it's checking to see if the newProps.selectedDate is not equal to this.props.selectedDate?

Comment (cerinman) [inline src/components/Calendar.js]: Copied the code over from the DatePicker component so I can't say for sure but it appears that whoever wrote it was trying to protect against undefined errors.

**Comment by cerinman** [comment]
Not sure either @iheartrachie but I'm going to try and make it work somehow so we don't have to maintain multiple calendars.  I'll keep you posted.

**Comment by redperk** [comment]
:shipit:

**Pull request merged**